.github/workflows/publish.yml

```diff
@@ -5,26 +5,40 @@ permissions:
 jobs:
   publish:
     runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+      NPM_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
     steps:
       - name: Check out code
         uses: actions/checkout@v3
+
       - name: Setup job
         uses: ./.github/actions/setup-job
+
       - name: Set git config
         run: |
           git config --global user.email "jackblakeston93@gmail.com"
           git config --global user.name "Jack Blakeston"
-      - name: Bump and Publish changed packages
-        env:
-          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
-          NPM_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+      - name: Install Lerna
+        run: yarn global add lerna@^6.5.1
+
+      - name: Bump package versions
         run: |
-          yarn global add lerna@^6.5.1
           if lerna changed --json --loglevel=error | grep -qF "packages/"; then
-            echo "//registry.npmjs.org/:_authToken=\${NPM_AUTH_TOKEN}" > ~/.npmrc
             yarn lerna version --yes --conventional-commits --no-changelog
```

Comment:
These changes generate new versions of the components I imagine, but then the package.json with the new versions are comitted and pushed? How will it work in the end?

Reply:
Let's merge this branch and you can see it

Reply:
Although Im a bit worried that the main branch might have some protection that impedes this from working, that's why i want to merge so we can test that